onnx/numpy_helper.py

```diff
@@ -272,6 +272,67 @@ def from_dict(dict, name=None):  # type: (Dict[Any, Any], Optional[Text]) -> Map
     return map
 
 
+def to_optional(optional):  # type: (OptionalProto) -> Optional[Any]
+    """Converts an optional def to a Python list.
+
+    Inputs:
+        optional: an OptionalProto object.
+    Returns:
+        lst: the converted list.
+    """
+    opt = None  # type: Optional[Any]
+    elem_type = optional.elem_type
+    value_field = mapping.STORAGE_ELEMENT_TYPE_TO_FIELD[elem_type]
+    values = getattr(optional, value_field)
+    for value in values:
+        if elem_type == OptionalProto.TENSOR or elem_type == OptionalProto.SPARSE_TENSOR:
+            opt = to_array(value)
+        elif elem_type == OptionalProto.SEQUENCE:
+            opt = to_list(value)
+        elif elem_type == OptionalProto.MAP:
+            opt = to_dict(value)
+        else:
+            raise TypeError("The element type in the input optional is not supported.")
```

Comment:
Random thought:
Is it possible to create a new utility function for these if-else conditions? (L288-L295 and L150-L157 is a similar logic) `from_list` and `from_optional` have the same situation.

Since using another new function might lose some readability, honestly I am open to both options (keep the original way or introduce a new function). Thanks.

Reply:
I think we can have two maps and use the key of elem_type to get corresponding the transform. I think this is pretty readable and small for now. I'll leave a todo if the list gets expanded.